server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsResponse.java

```diff
@@ -51,24 +52,31 @@
             (p, c) -> p.map(HashMap::new, ElasticsearchException::fromXContent),
             new ParseField("failures")
         );
+        GET_SNAPSHOT_PARSER.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), new ParseField("next"));
     }
 
     private final List<SnapshotInfo> snapshots;
 
     private final Map<String, ElasticsearchException> failures;
 
-    public GetSnapshotsResponse(List<SnapshotInfo> snapshots, Map<String, ElasticsearchException> failures) {
+    @Nullable
+    private final String next;
+
+    public GetSnapshotsResponse(List<SnapshotInfo> snapshots, Map<String, ElasticsearchException> failures, @Nullable String next) {
         this.snapshots = List.copyOf(snapshots);
         this.failures = failures == null ? Map.of() : Map.copyOf(failures);
+        this.next = next;
     }
 
     public GetSnapshotsResponse(StreamInput in) throws IOException {
         this.snapshots = in.readList(SnapshotInfo::readFrom);
         if (in.getVersion().onOrAfter(GetSnapshotsRequest.MULTIPLE_REPOSITORIES_SUPPORT_ADDED)) {
```

Comment:
I take it you will backport both this and the multiple repo PR in one go then?

Reply:
Depends on whether this goes in today I guess with FF incoming :) If not I'll adjust the PR and backport separately.